Lombiq.Tests.UI/Services/SqlServerManager.cs

```diff
@@ -85,39 +90,46 @@ public SqlServerRunningContext CreateDatabase()
+    /// <paramref name="snapshotDirectoryPathLocal"/> should be the same or the latter can be left at the default
+    /// <see langword="null"/> value. If the server is on a remote location or within a container, then the
+    /// <paramref name="snapshotDirectoryPathRemote"/> directory must be mounted to the
+    /// <paramref name="snapshotDirectoryPathLocal"/> location on the local file system. For example via a mounted
+    /// volume on Docker or an (S)FTP network location mounted as a drive.
     /// </summary>
-    /// <param name="snapshotDirectoryPathRemote">The location of the save directory on the SQL Server's machine.</param>
+    /// <param name="snapshotDirectoryPathRemote">
+    /// The location of the save directory on the SQL Server's machine.
+    /// </param>
     /// <param name="snapshotDirectoryPathLocal">
-    /// The location of the directory where the saved database snapshot can be accessed from the local system. If <see
-    /// langword="null"/>, it takes on the value of <paramref name="snapshotDirectoryPathRemote"/>. Useful if for local
-    /// server only calls.
+    /// The location of the directory where the saved database snapshot can be accessed from the local system. If
+    /// <see langword="null"/>, it takes on the value of <paramref name="snapshotDirectoryPathRemote"/>. Useful if
+    /// for local server only calls.
     /// </param>
     /// <param name="useCompressionIfAvailable">
-    /// If set to <see langword="true"/> and the database engine supports it, then <see
-    /// cref="BackupCompressionOptions.On"/> will be used.
+    /// If set to <see langword="true"/> and the database engine supports it, then
+    /// <see cref="BackupCompressionOptions.On"/> will be used.
     /// </param>
-    public void TakeSnapshot(
+    public async Task TakeSnapshotAsync(
         string snapshotDirectoryPathRemote,
         string snapshotDirectoryPathLocal = null,
+        string containerName = null,
         bool useCompressionIfAvailable = false)
     {
         var filePathRemote = GetSnapshotFilePath(snapshotDirectoryPathRemote);
         var filePathLocal = GetSnapshotFilePath(snapshotDirectoryPathLocal ?? snapshotDirectoryPathRemote);
+        var directoryPathLocal =
+            Path.GetDirectoryName(filePathLocal) ??
+            throw new InvalidOperandException($"Failed to get the directory path for local path \"{filePathLocal}\".");
 
+        if (!Directory.Exists(directoryPathLocal)) Directory.CreateDirectory(directoryPathLocal);
         if (File.Exists(filePathLocal)) File.Delete(filePathLocal);
 
         var server = CreateServer();
 
         KillDatabaseProcesses(server);
 
         var useCompression = useCompressionIfAvailable &&
-            (server.EngineEdition == Edition.EnterpriseOrDeveloper || server.EngineEdition == Edition.Standard);
+                             (server.EngineEdition == Edition.EnterpriseOrDeveloper || server.EngineEdition == Edition.Standard);
```

Comment:
That's a looong indentation.

Reply:
It wants to indent it to the same column where the previous expression starts (so after the `=`). That's why you have to break the first part into a new line as well:
```csharp
        var useCompression =
            useCompressionIfAvailable &&
            (server.EngineEdition == Edition.EnterpriseOrDeveloper || server.EngineEdition == Edition.Standard);
```

Reply:
Yeah but adjusting to the column like this is something we never do (for a reason) though I know you like it ;). I'm OK with the second version too.